### PR TITLE
refactor: input handling

### PR DESCRIPTION
**Pull request by AnthonyWadham**
# Add input client for interactive CLI prompts

Closes: WORLD-XXX

## Overview

This PR adds a new input client package to handle interactive CLI prompts in a testable way. The implementation provides methods for prompting users for text input, confirmation (Y/n), and selection from a list of options.

## Brief Changelog

- Created a new `input` package with a client interface for handling user input
- Implemented methods for text prompts, confirmations, and option selection
- Added support for default values and context cancellation
- Created mock and test implementations for testing
- Integrated the input client with the setup service, replacing the hardcoded `getInput` function
- Updated organization and project handlers to use the new input client

## Testing and Verifying

This change added tests and can be verified as follows:
- Added mock implementations for unit testing
- Added a test client for integration testing
- The implementation handles various edge cases like default values, cancellation, and invalid inputs

## Greptile Summary

Introduces a new input client package to handle interactive CLI prompts, improving testability and maintainability of user interactions throughout the codebase.

- Added `cmd/pkg/clients/input` package with ClientInterface for text prompts, confirmations, and option selection
- Integrated input client with setup service, replacing hardcoded `getInput` functions in organization and project handlers
- Added mock implementations (MockClient and TestInputClient) for testing with predefined responses
- Improved error handling with proper context cancellation support across input operations
- Implemented default value support and validation for different input types (text, Y/n, selections)

### CODE REVIEW

**Comment by Doglightning** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/218?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#219** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/219?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#218** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/218?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/218?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#217** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/217?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#216** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/216?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#215** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/215?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#214** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/214?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite/merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/Argus-Labs?ref=merge-queue-instructions-in-stack-comment&prId=218).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: cmd/pkg/clients/input/client.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+				c.println("Invalid input. Please enter 'y' or 'n'")
+				continue
+			}
+		}
+	}
+}
+
+// Select allows user to select from multiple options by number.
+//
+//nolint:gocognit // This function is complex but seperated into smaller functions would make it harder to read.
```
Comment (ezavada) [inline cmd/pkg/clients/input/client.go:109]: separated, with an "a"

File: cmd/pkg/clients/input/client.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	"pkg.world.dev/world-cli/common/printer"
+)
+
+var (
+	ErrInputCanceled = eris.New("input canceled")
+	ErrInvalidInput  = eris.New("invalid input")
+)
+
+// NewClient creates a new input client with standard stdin/stdout.
+func NewClient() Client {
```
Comment (ezavada) [inline cmd/pkg/clients/input/client.go:22]: Nit: To me, a client is something that talks to a remote service. So I don't love this terminology. I'd prefer it be considered a Service.

**Review by ezavada — APPROVED** [review]
